tests/Custom/CapabilityTest.php

```diff
@@ -0,0 +1,103 @@
+            'browserName' => 'fake browser',
+            'desiredCapabilities' => [],
+            'expectedCapabilities' => [
+                'platform' => 'ANY',
+                'name' => 'Behat Test',
+                'deviceOrientation' => 'landscape',
+                'deviceType' => 'desktop',
+            ],
+        ];
+
+        yield 'default capabilities can be customised' => [
+            'browserName' => 'fake browser',
+            'desiredCapabilities' => [
+                'something' => 'custom',
+                'name' => 'Custom Test',
+            ],
+            'expectedCapabilities' => [
+                'platform' => 'ANY',
+                'name' => 'Custom Test',
+                'deviceOrientation' => 'landscape',
+                'deviceType' => 'desktop',
+                'something' => 'custom',
+            ],
+        ];
+
+        yield 'browser-specific default capabilities are added' => [
+            'browserName' => 'chrome',
+            'desiredCapabilities' => [],
+            'expectedCapabilities' => [
+                'browserName' => 'chrome',
+                'platform' => 'ANY',
+                'name' => 'Behat Test',
+                'deviceOrientation' => 'landscape',
+                'deviceType' => 'desktop',
+                'goog:chromeOptions' => [
+                    'excludeSwitches' => ['enable-automation'],
+                ],
+            ],
+        ];
+
+        yield 'browser-specific default capabilities can be customised' => [
+            'browserName' => 'chrome',
+            'desiredCapabilities' => [
+                'name' => 'Custom Test',
+                'goog:chromeOptions' => ['args' => ['a', 'b', 'c']],
+            ],
+            'expectedCapabilities' => [
+                'browserName' => 'chrome',
+                'platform' => 'ANY',
+                'name' => 'Custom Test',
+                'deviceOrientation' => 'landscape',
+                'deviceType' => 'desktop',
+                'goog:chromeOptions' => ['args' => ['a', 'b', 'c']],
+            ],
+        ];
+    }
+
+    /**
+     * @param string $browserName
+     * @param array<string, mixed> $desiredCapabilities
+     * @return WebdriverClassicDriver&object{capabilities: null|array<string, mixed>}
+     */
+    private function createDriverExposingCapabilities(string $browserName, array $desiredCapabilities = []): WebdriverClassicDriver
+    {
+        return new class($browserName, $desiredCapabilities) extends WebdriverClassicDriver {
+            /**
+             * @var null|array<string, mixed>
+             */
+            public ?array $capabilities = null;
+
+            public function __construct(string $browserName, array $desiredCapabilities)
+            {
+                parent::__construct($browserName, $desiredCapabilities);
+
+                $this->capabilities = $this->getDesiredCapabilities();
+            }
```

Comment:
Or you can just make the `getDesiredCapabilities` method public in this extended class.

Reply:
True, that should also work.. 🤔

Reply:
Ah, I know why I did it like that - it's because in PHPStan, I cannot type hint a method on an anonymous object (see line 85).

Reply:
OK.